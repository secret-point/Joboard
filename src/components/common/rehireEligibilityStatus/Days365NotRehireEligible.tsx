import React from "react";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import { translate as t } from "../../../utils/translator";
import { Job } from "../../../utils/types/common";

interface Days365NotRehireEligibleProps {
  jobDetail?: Job
}

const Days365NotRehireEligible = (props: Days365NotRehireEligibleProps ) => {

  const { jobDetail } = props;

  const tileWithJobTitle = t("BB-365-days-not-rehire-eligible-title-text-with-jobTitle", "Thank you for your interest in the {jobTitle} position. Our records indicate you previously worked at Amazon and are not eligible for rehire based upon your previous Amazon employment record. As a result, we are unable to move forward with your candidacy at this time.", {jobTitle: jobDetail?.jobTitle || ""});
  const titleWithoutJobTitle = t("BB-365-days-not-rehire-eligible-title-text-without-jobTitle", "Thank you for your interest in this position. Our records indicate you previously worked at Amazon and are not eligible for rehire based upon your previous Amazon employment record. As a result, we are unable to move forward with your candidacy at this time.")

  return (
    <Col gridGap="S300">
      <Text>
        {
          jobDetail ? tileWithJobTitle : titleWithoutJobTitle
        }
      </Text>
      <Text>
        {t("BB-365-days-not-rehire-eligible-content-text", "Due to the termination reason, you can reapply 90 days after your last day worked at Amazon.")}
      </Text>
    </Col>
  )
}

export default Days365NotRehireEligible;
