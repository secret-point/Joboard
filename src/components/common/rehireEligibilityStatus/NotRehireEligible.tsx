import React from "react";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import { translate as t } from "../../../utils/translator";

const NotRehireEligible = () => {
  return (
    <Col gridGap="S300">
      <Text>
        {t("BB-not-rehire-eligible-title-text", "Thank you for your recent job application to Amazon. After careful consideration, we will not be moving forward with your job application.")}
      </Text>
      <Text>
        {t("BB-not-rehire-eligible-content-text", "We value your time and wish you the best of luck in the future.")}
      </Text>
    </Col>
  );
};

export default NotRehireEligible;
