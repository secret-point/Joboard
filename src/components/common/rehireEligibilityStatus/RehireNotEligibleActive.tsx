import React from "react";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import { translate as t } from "../../../utils/translator";

const RehireNotEligibleActive = () => {
  return (
    <Col gridGap="S300">
      <Text>
        {t("BB-rehire-not-eligible-active-title-text", "We appreciate your recent job application! Unfortunately, you are not eligible for this particular position since you are already employed by Amazon.")}
      </Text>
      <Text>
        {t("BB-rehire-not-eligible-active-text-open-opportunities-list-title", "The good news? There are plenty of open opportunities that you can browse and apply for online. Here are a few of the ways you can explore your future career growth:")}
      </Text>
      <ul>
        <li>
          <Text>
            {t("BB-rehire-not-eligible-active-text-open-opportunities-list-item-one", "For Agency or Seasonal conversions, visit amazon.com/conversions (http://amazon.com/conversions)")}
          </Text>
        </li>
        <li>
          <Text>
            {t("BB-rehire-not-eligible-active-text-open-opportunities-list-item-two", "To request shift or location changes, reach out to your onsite team")}
          </Text>
        </li>
      </ul>
      <Text>
        {t("BB-rehire-not-eligible-active-content-text-one", "Your local HR team or the Employee Resource Center (ERC) are the best resource for any questions. You can reach the ERC team 24/7 at 1-888-892-7180 (tel:18888927180).")}
      </Text>
      <Text>
        {t("BB-rehire-not-eligible-active-content-text-two", "Thanks for being a part of the amazing team at Amazon! We are continually striving to provide you with the best possible employee resources, benefits, pay and flexible schedules — all supporting our efforts to become Earth’s Greatest Employer!",)}
      </Text>
    </Col>
  );
};

export default RehireNotEligibleActive;
