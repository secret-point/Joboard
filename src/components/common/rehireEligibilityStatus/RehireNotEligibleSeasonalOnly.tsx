import React from "react";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import { translate as t } from "../../../utils/translator";
import { Job } from "../../../utils/types/common";

interface RehireNotEligibleSeasonalOnlyProps {
  jobDetail?: Job
}

const RehireNotEligibleSeasonalOnly = (props: RehireNotEligibleSeasonalOnlyProps) => {

  const { jobDetail } = props;

  const titleWithJobTitle = t("BB-rehire-not-eligible-seasonal-only-title-text-with-jobTitle", "Thank you for your interest in the {jobTitle} position. Our records indicate that you previously worked at Amazon.  At this time, you are eligible for rehire as a Seasonal Amazon Associate or third-party agency employee based upon your employment record. As a result, we are unable to move forward with your candidacy for this job. If you have any questions regarding rehire eligibility, contact the site HR Team.", {jobTitle: jobDetail?.jobTitle || ""});
  const titleWithoutJobTitle = t("BB-rehire-not-eligible-seasonal-only-title-text-without-jobTitle", "Thank you for your interest in this position. Our records indicate that you previously worked at Amazon.  At this time, you are eligible for rehire as a Seasonal Amazon Associate or third-party agency employee based upon your employment record. As a result, we are unable to move forward with your candidacy for this job. If you have any questions regarding rehire eligibility, contact the site HR Team.");

  return (
    <Col gridGap="S300">
      <Text>
        {
          jobDetail?.jobTitle ? titleWithJobTitle : titleWithoutJobTitle
        }
      </Text>
      <Text>
        {t("BB-rehire-not-eligible-seasonal-only-content-text", "Please visit our Seasonal Page to explore seasonal opportunities in your area.")}
      </Text>
    </Col>
  )
}

export default RehireNotEligibleSeasonalOnly;
