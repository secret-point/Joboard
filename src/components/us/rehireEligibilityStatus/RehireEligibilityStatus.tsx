import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { Col } from "@amzn/stencil-react-components/layout";
import React, { useEffect } from "react";
import { connect } from "react-redux";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  redirectToDashboard,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { translate as t } from "../../../utils/translator";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { checkAndBoundGetApplication, getLocale } from "../../../utils/helper";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { useLocation } from "react-router-dom";
import queryString from "query-string";
import { JobState } from "../../../reducers/job.reducer";
import { WorkflowState } from "../../../reducers/workflow.reducer";
import { WORKFLOW_ERROR_CODE } from "../../../utils/enums/common";
import RehireNotEligibleActive from "../../common/rehireEligibilityStatus/RehireNotEligibleActive";
import Days365NotRehireEligible from "../../common/rehireEligibilityStatus/Days365NotRehireEligible";
import RehireNotEligibleSeasonalOnly from "../../common/rehireEligibilityStatus/RehireNotEligibleSeasonalOnly";
import NotRehireEligible from "../../common/rehireEligibilityStatus/NotRehireEligible";
import { ApplicationState } from "../../../reducers/application.reducer";

interface MapStateToProps {
  candidate: CandidateState,
  job: JobState,
  workflow: WorkflowState,
  application: ApplicationState
}

export const RehireEligibilityStatus = (props: MapStateToProps) => {
  const { job, workflow, candidate, application } = props;
  const { search, pathname } = useLocation();
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const pageName = getPageNameFromPath(pathname);
  const { jobId, applicationId } = queryParams;
  const jobDetail = job.results;
  const applicationData = application.results;
  const candidateData = candidate.results.candidateData;
  const { workflowErrorCode } = workflow;

  useEffect(() => {
    boundGetCandidateInfo();
  }, [])

  useEffect(() => {
    checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() })
  }, [jobId]);

  useEffect(() => {
    // Page will emit page load event once both pros are available but
    // will not emit new event on props change once it has emitted pageload event previously
    jobDetail && applicationData && candidateData && addMetricForPageLoad(pageName);

  }, [jobDetail, applicationData, candidateData]);

  useEffect(() => {
    return () => {
      //reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    }
  },[]);

  const handleGoToDashboard = () => {
    redirectToDashboard();
  }

  const renderRehireEligibility = () => {
    switch (workflowErrorCode) {
      case WORKFLOW_ERROR_CODE.ACTIVE:
        return <RehireNotEligibleActive/>

      case WORKFLOW_ERROR_CODE.NOT_REHIRE_ELIGIBLE_365_DAYS:
        return <Days365NotRehireEligible/>;

      case WORKFLOW_ERROR_CODE.SEASONAL_ONLY:
        return <RehireNotEligibleSeasonalOnly/>;

      case WORKFLOW_ERROR_CODE.NOT_REHIRE_ELIGIBLE:
        return <NotRehireEligible/>;

      default:
        return <NotRehireEligible/>;
    }
  }

  return (
    <Col gridGap="S300" padding={{ top: 'S300' }}>
      {
        renderRehireEligibility()
      }
      <Col padding={{top: "S300"}}>
        <Button variant={ButtonVariant.Primary} onClick={handleGoToDashboard}>
          {t("BB-rehire-eligibility-status-back-to-dashboard-button-text", "Return to dashboard")}
        </Button>
      </Col>
    </Col >
  )
}

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(RehireEligibilityStatus);
