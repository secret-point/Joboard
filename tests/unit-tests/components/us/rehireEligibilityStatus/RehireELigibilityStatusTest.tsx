import React from "react";
import { shallow } from "enzyme";
import routeData from 'react-router';
import { RehireEligibilityStatus } from "../../../../../src/components/us/rehireEligibilityStatus/RehireEligibilityStatus";
import { TEST_APPLICATION_ID, TEST_APPLICATION_STATE, TEST_CANDIDATE_STATE, TEST_JOB_ID, TEST_JOB_STATE, TEST_WORKFLOW_STATE } from "../../../../test-utils/test-data";

describe("RehireEligibilityStatus", () => {
  const mockLocation = {
    pathname: "/rehire-eligibility-status",
    search: `?jobId=${TEST_JOB_ID}&applicationId=${TEST_APPLICATION_ID}`,
    hash: '',
    state: null
  };

  it("should match snapshot", () => {
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);

    const shallowWrapper = shallow(
      <RehireEligibilityStatus
        candidate={TEST_CANDIDATE_STATE}
        job={TEST_JOB_STATE}
        application={TEST_APPLICATION_STATE}
        workflow={TEST_WORKFLOW_STATE}
      />);

    expect(shallowWrapper).toMatchSnapshot();
  });
});
