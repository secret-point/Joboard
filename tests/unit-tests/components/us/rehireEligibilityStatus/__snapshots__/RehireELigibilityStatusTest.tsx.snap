// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RehireEligibilityStatus should match snapshot 1`] = `
<Col
  gridGap="S300"
  padding={
    Object {
      "top": "S300",
    }
  }
>
  <RehireNotEligibleActive />
  <Col
    padding={
      Object {
        "top": "S300",
      }
    }
  >
    <Button
      colorScheme="default"
      iconPosition="leading"
      onClick={[Function]}
      shouldUseResponsiveContent={true}
      size="default"
      type="button"
      variant="primary"
    >
      Return to dashboard
    </Button>
  </Col>
</Col>
`;
